treesource.c

```diff
@@ -139,26 +139,49 @@ static const char *delim_end[] = {
 	[TYPE_STRING] = "",
 };
 
+/*
+ * The invariants in the marker list are:
+ *  - offsets are monotonically rising
```

Comment:
Maybe "non-strictly monotonically increasing", since adjacent offsets can be equal.